src/studio/nuxeo-studio/working-in-studio/migration-services.md

```diff
@@ -0,0 +1,120 @@
+
+{{/callout}}
+
+### Launching Migration
+
+If your project benefit from branch management: In **Source Control**, go to **Branch Management**, checkout the branch you want to migrate.
+
+
+1. In **Configuration**, go to **Migration Services**.
+1. Click on **Start Migration**.
+A pop-up appears to ensure you want to start migrating.
+1. Click **OK** to confirm.
+
+![]({{file name='migration-services-jsf-to-web-ui-screen.png'}} ?w=650,border=true)
+
+{{#> callout type='note' }}
+
+If you are using the intermediate or advanced commit modes, you need to discard or commit and push any work in progress before the migration can be launched.
+
+{{/callout}}
+
+Migration can be launched as many times as you wish: for example, to allow you to test changes made on your JSF UI configuration or benefit from new features that have been added to the migration tool.
+
+{{#> callout type='warning' }}
+
+The branch created for the migration will be deleted and created again with every migration attempt, resulting in the loss of any changes made into it that would have not been merged yet.
+
+{{/callout}}
+
+### Reviewing Changes
+
+When the migration is finished, a report is available in **Migration Services**.
+
+![]({{file name='migration-services-access-report.png'}} ?w=650,border=true)
+
+Click **Access Report** to see and easily search through the result in order to find any particular information.
+
+
+The report provides a summary of the migration. Details for each component of your project are displayed, with descriptions and recommendations when errors occurred.
+
+It also provides information concerning the migration itself, such as the tool version, source branch, result branch, source tag, destination tag and date.
+
+### Integrating Changes
+
+Results appear in a branch indicated in the **Previous Migration** list.
+
+To check the result of your migration, go to  **Source Control** and **Branch Management**.  
+
+To switch to the new branch, press **Checkout** on it.
+
+The migration process does not delete any configuration but it adds more configuration that might be slightly updated. For example, to enforce a good practice such as putting constraints expressed only in your layouts back at schema level. The configuration produced can be reviewed and changed if needed, at your own pace.
+
+To merge the changes, follow the usual [feature implementation process]({{page page="how-to-implement-features-using-branches"}}). If your project benefits from branch management features, you can also take advantage of the [configuration review process]({{page page="how-to-review-nuxeo-studio-configuration"}}) process to ease the merge.
+
+The JSF UI configuration can be hidden when you are ready to migrate. For more information, see the finalizing migration section.
+
+### Discarding Changes
+
+If your project benefits from branch management, click on **Settings**, go to **Branch Management** and then delete the branch created by the migration.
```

Comment:
This is wrong actually, we create a branch even if you don't have branch management now.

Difference is if you have branch management we'll create a new branch every time (nothing gets erased). If you don't we always use the same branch, it's deleted and created again with every migration so you lose its previous content in the process.

Reply:
integrated :) (many thanks for your crucial feedback)